packages/embeds/embed-core/src/embed-iframe.ts

```diff
@@ -2,18 +2,40 @@ import { useRouter } from "next/router";
 import type { CSSProperties } from "react";
 import { useState, useEffect } from "react";
 
+import type { Message } from "./embed";
 import { sdkActionManager } from "./sdk-event";
 
-export interface UiConfig {
+export type UiConfig = {
   hideEventTypeDetails?: boolean;
   theme?: "dark" | "light" | "auto";
-  styles?: EmbedStyles;
-}
+  styles?: EmbedStyles & EmbedNonStylesConfig;
+};
+
+type SetStyles = React.Dispatch<React.SetStateAction<EmbedStyles>>;
+type setNonStylesConfig = React.Dispatch<React.SetStateAction<EmbedNonStylesConfig>>;
+
+const embedStore = {
+  // Store all embed styles here so that as and when new elements are mounted, styles can be applied to it.
+  styles: {} as EmbedStyles | undefined,
+  nonStyles: {} as EmbedNonStylesConfig | undefined,
+  namespace: null as string | null,
+  embedType: undefined as undefined | null | string,
+  // Store all React State setters here.
+  reactStylesStateSetters: {} as Record<keyof EmbedStyles, SetStyles>,
+  reactNonStylesStateSetters: {} as Record<keyof EmbedNonStylesConfig, setNonStylesConfig>,
+  parentInformedAboutContentHeight: false,
+  windowLoadEventFired: false,
+  setTheme: undefined as ((arg0: string) => void) | undefined,
+  theme: undefined as UiConfig["theme"],
+  uiConfig: undefined as Omit<UiConfig, "styles" | "theme"> | undefined,
+  setUiConfig: undefined as ((arg0: UiConfig) => void) | undefined,
+};
```

Comment:
Instead of doing assertion on entire object, do it per property, so that if a property is missing, it can be identified by typescript.